Signup: Show domain step in import flows

creativecoder opened the pull request:
#### Changes proposed in this Pull Request

* Show domain step in `import` and `import-onboarding` flows (no longer autocomplete it)
* Autofill domain input with a suggested domain import site url
    - If there's a path in the url, such as `https://hi6822.wixsite.com/eat-here-its-good`, suggest the path `eat-here-its-good`
    - Else if its a WordPress.com, Blogger, GoDaddy, Squarespace, or Tumblr site, such as `http://theperfectrealestate.godaddysites.com/`, suggest the subdomain `theperfectrealestate`
    - Otherwise suggest the domain, like `example.com` or `blog.example.com`

<img width="1237" alt="Screen Shot 2019-08-07 at 17 12 18" src="https://user-images.githubusercontent.com/1699996/62661417-9cb52300-b936-11e9-89c1-810aae61224e.png">


#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

##### Import flow

- Visit `/start/import`
- Complete the user step, if logged out
- Enter a Wix or GoDaddy site url
- When you get to the domains step, you should see a suggested domain created from the site url you entered in the previous step and a search for an available domain should start automatically
- Select a free or paid domain and continue, confirming that you can complete the flow and create a new site

##### Domains flow

Domains only flows also search automatically in the domains step, so verify that's still working

- Visit `/start/domain/domain-only?new=testtesttesttest&search=yes`
- Verify that `testtesttesttest` is filled in the search input a search for an available domain starts automatically

Commit:
Show domain step in import flows (reverts changes in #27401)

Commit:
Suggests domain from import url

Commit:
Uses signup dependency to provide a suggested domain for imported sites

Commit:
Updates comments

Commit:
Updates import in signup e2e test to check for suggested domain

Commit:
Correct comment

creativecoder commented:
Pinging @aidvu , who worked on domains-only auto searching, to make sure I haven't missed anything else related to this functionality.

aidvu commented on client/signup/steps/domains/index.jsx:
```diff
@@ -85,27 +85,13 @@ class DomainsStep extends React.Component {
-
 		const domain = get( props, 'queryObject.new', false );
 		const search = get( props, 'queryObject.search', false ) === 'yes';
+		const suggestedDomain = get( props, 'signupDependencies.suggestedDomain' );
```
Doing things in the constructor like this is very brittle. Please test all the cases when landing on the domain screen with the `suggestedDomain` set like refreshing the browser, going back and forth through the steps.

andfinally commented on client/lib/importers/utils.ts:
```diff
@@ -0,0 +1,29 @@
+	}
+
+	// Other services use a subdomain, like `example.wordpress.com.
+	if ( parsedHostname && parsedHostname.length > 2 ) {
```
Just discovered `url.parse` ignores the `www`, so this works fine!

andfinally reviewed (commented):
This LGTM, I've tested with a variety of URLs in both logged-out and logged-in states, and using the browser back button and flow back button, and everything works as expected. We're getting eight tests failing - I'm not sure why, for example:

```
"after all" hook - [WPCOM] Calypso Gutenberg Editor: Pages (desktop) Insert a payment button into a page: @parallel
invalid session id

"after all" hook - [WPCOM] Editor: Pages (desktop) Insert a payment button into a page: @parallel @jetpack
invalid session id

"after all" hook - [WPCOM] Editor: Posts (desktop) Insert a payment button: @parallel @jetpack
invalid session id
```

andfinally reviewed (approved):
Tests have now passed, so apart from resolving the conflict I'd say this is good to go!

Commit:
Restores generation of a suggested domain lost in rebase

Commit:
Updates suggested domains for selected importer types.

creativecoder commented on client/lib/signup/step-actions.js:
```diff
@@ -192,6 +185,11 @@ export function createSiteWithCart(
 		newSiteParams.find_available_url = !! isPurchasingItem;
 	}
 
+	if ( 'import' === lastKnownFlow || 'import-onboarding' === lastKnownFlow ) {
```
I'm not totally sure either, but in testing adding the plans step to the `import` flow, we saw `flowName` empty in some cases, so it seems more reliable to use `lastKnownFlow`

Commit:
Corrects typo in e2e step name

creativecoder commented:
I've rebased this and updated the domain suggestion logic. Now subdomain is used as a suggestion only for specific services, WordPress.com, Blogger, GoDaddy, Squarespace, and Tumblr. Otherwise we suggest the full domain, so the suggestion for `blog.example.com` isn't `blog`, it's `blog.example.com`.

jonathansadowski reviewed (changes requested):
Can we move the `client/lib/importers` code into `client/lib/importer`? I'm attempting to make that change in #35240, as I feel the distinction between the two isn't strong enough to warrant the two confusingly similar names

jonathansadowski reviewed (changes requested):
Edit: False alarm, testing error.

Commit:
Moves import utils into lib/importer to consolidate files

creativecoder commented:
> Can we move the client/lib/importers code into client/lib/importer? I'm attempting to make that change in #35240, as I feel the distinction between the two isn't strong enough to warrant the two confusingly similar names

Good call -- updated.

jonathansadowski reviewed (approved):
Works well for me. Thanks for making the change to the library path! Goodjob
